src/plugins/syntax-highlighting/bracket-pair-colorization.ts

```diff
@@ -0,0 +1,47 @@
+export function generateBracketPairColorizationCSS(
+  colors: [number, number, number][],
+  colorInText: boolean,
+  thickenBrackets: number
+) {
+  if (!colors[0]) return [];
+
+  const colorMaker = `rgb(${colors[0]
+    .map((_, i) => {
+      return `calc(${colors
+        .map((col, colindex) => {
+          const channel = col[i];
+          return `${channel} * max(0, 10000 * cos(
+            2 * 3.14159265358979323 * (var(--test2) - ${colindex}) / ${colors.length}
+            ) - 9999)`;
+        })
+        .join(" + ")})`;
+    })
+    .join(", ")})`;
```

Comment:
This took a bit for me to figure out. Might want to explain how the periodic cosine effects the color cycling. The scale to [-19999,1] (then clamping to non-negative) is used for what? Ensuring that anything slightly off-peak is zero?

Reply:
That's correct. I'll add a comment clarifying, nice catch.

Reply:
Done.